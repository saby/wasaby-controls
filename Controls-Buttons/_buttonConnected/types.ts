import type { ObjectMeta } from 'Types/meta';

type TCommandOptions = Record<string, unknown>;

/**
 * Интерфейс объекта, описывающего действия.
 * @public
 */
export interface IActionConfig<RuntimeInterface extends object> {
    /**
     * Уникальный идентификатор действия
     * @remark это значение пишется в настройки действия во фрейме
     */
    id: string;

    /**
     * Человекопонятное название действия
     */
    title: string;
    /**
     * Категория действия
     */
    category?: string;
    /**
     * Иконка действия
     */
    icon?: string;
    /**
     * Алиас до модуля-исполнителя команды
     * @interface
     */
    commandName?: string;
    /**
     * Опции, которые пережаются в commandName первым аргументом
     */
    commandOptions?: TCommandOptions;
    /**
     * Мета-тип атрибутов команды
     */
    propsMeta?: ObjectMeta<RuntimeInterface>;
}

/**
 * Интерфейс модуля-исполнителя команды
 */
export interface IActionExecutor {
    /**
     * Метод, запускающий выполнение действия
     * @param commandOptions опции действия, заданные в его описании
     * @param editorProps опции действия, настроенные в конструкторе через {@link Controls-editors/propertyGrid:PropertyGrid PropertyGrid}
     */
    execute(commandOptions: TCommandOptions, editorProps: Record<string, unknown>);
}
