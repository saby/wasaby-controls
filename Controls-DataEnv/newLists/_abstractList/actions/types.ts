import type * as marker from './types/marker';
import type * as operationsPanel from './types/operationsPanel';
import type * as root from './types/root';
import type * as search from './types/search';
import type * as selection from './types/selection';
import type * as source from './types/source';
import type * as expandCollapse from './types/expandCollapse';
import type * as filter from './types/filter';

import type * as interactorCore from './types/interactorCore';

export type {
    marker,
    operationsPanel,
    root,
    search,
    selection,
    source,
    expandCollapse,
    filter,
    interactorCore,
};

/**
 * Тип действия, доступного в любом списке, независимо от типа ViewModel, к которой он подключен (web/mobile).
 */
export type TAnyAbstractAction =
    | source.TAnySourceAction
    | expandCollapse.TAnyExpandCollapseAction
    | marker.TAnyMarkerAction
    | filter.TAnyFilterAction
    | selection.TAnySelectionAction
    | operationsPanel.TAnyOperationsPanelAction
    | root.TAnyRootAction
    | search.TAnySearchAction
    | interactorCore.TAnyInteractorCoreAction;
