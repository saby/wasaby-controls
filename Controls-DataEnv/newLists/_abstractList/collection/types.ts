export { TCollectionType } from './types/TCollectionType';
