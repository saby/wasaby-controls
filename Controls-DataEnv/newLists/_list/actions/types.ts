import type { TAbstractListActions } from 'Controls-DataEnv/abstractList';

import type * as marker from './types/marker';
import type * as operationsPanel from './types/operationsPanel';
import type * as root from './types/root';
import type * as search from './types/search';
import type * as selection from './types/selection';
import type * as items from './types/items';
import type * as filter from './types/filter';
import type * as source from './types/source';
import type * as expandCollapse from './types/expandCollapse';
import type * as complexUpdate from './types/complexUpdate';

export type {
    marker,
    operationsPanel,
    root,
    expandCollapse,
    search,
    selection,
    items,
    filter,
    source,
    complexUpdate,
};

/**
 * Тип действия, доступного в WEB списке.
 */
export type TAnyListAction =
    | TAbstractListActions.TAnyAbstractAction
    | expandCollapse.TAnyExpandCollapseAction
    | marker.TAnyMarkerAction
    | selection.TAnySelectionAction
    | operationsPanel.TAnyOperationsPanelAction
    | root.TAnyRootAction
    | search.TAnySearchAction
    | items.TAnyItemsAction
    | source.TAnySourceAction
    | filter.TAnyFilterAction
    | complexUpdate.TAnyComplexUpdateAction;
