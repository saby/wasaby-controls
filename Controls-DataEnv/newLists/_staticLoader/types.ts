import type { IAbstractListState } from 'Controls-DataEnv/abstractList';

export type TState = Pick<IAbstractListState, 'viewMode' | 'isLatestInteractorVersion'>;

export interface TDependencyDescriptor<T extends keyof TState> {
    prop: T;
    value: TState[T][] | ((state: TState) => boolean);
}

export type TUI_Dependencies = Record<string, TDependencyDescriptor<keyof TState>[]>;
