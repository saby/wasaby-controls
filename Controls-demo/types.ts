import { IHeaderCell } from 'Controls/grid';
import { RecordSet } from 'Types/collection';

export interface IHeader extends IHeaderCell {
    title: string;
}

export type TItemsReadyCallback = (items: RecordSet) => void;
export type TRoot = string | number | null;

export interface INavigation {
    source: string;
    view: string;
    sourceConfig: {
        pageSize: number;
        page: number;
        hasMore?: boolean;
    };
    viewConfig?: {
        pagingMode?: string;
        maxCountValue?: number;
    };
}
