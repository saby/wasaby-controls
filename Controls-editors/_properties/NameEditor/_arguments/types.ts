import { ObjectMeta } from 'Meta/types';

const PARAMETERS_FIELD = 'parameters';
const RETURN_VALUE_FIELD = 'return_value';

interface IFunctionCallMeta {
    arguments: ObjectMeta | null;
    returnValue: ObjectMeta | null;
}

export { IFunctionCallMeta, PARAMETERS_FIELD, RETURN_VALUE_FIELD };
