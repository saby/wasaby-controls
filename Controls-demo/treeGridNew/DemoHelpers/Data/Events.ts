import { IColumn } from 'Controls/grid';
import { IData } from '../Interface';

export const Events = {
    getColumns: (): IColumn[] => {
        return [
            {
                displayProperty: 'title',
                width: '',
            },
        ];
    },
    getDataBeforeExpanded: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataBeforeCollapsed: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataCollapsedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataExpandedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
};
