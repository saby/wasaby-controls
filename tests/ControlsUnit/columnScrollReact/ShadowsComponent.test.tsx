/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';

import * as React from 'react';

import ShadowsComponentConsumer, {
    ShadowsComponent,
} from 'Controls/_columnScrollReact/ShadowsComponent';
import {
    ColumnScrollContext,
    IColumnScrollContext,
} from 'Controls/_columnScrollReact/context/ColumnScrollContext';
import { SELECTORS } from './SelectorsForTests';
import { EdgeState } from 'Controls/_columnScrollReact/common/types';
import { renderAnyComponent } from './forTestsOnly/helpers';
import { AnimationSteps } from './forTestsOnly/animationSteps';

function FakeContextProviderForShadows(props: {
    children: JSX.Element;
    leftEdgeState: IColumnScrollContext['leftEdgeState'];
    rightEdgeState: IColumnScrollContext['rightEdgeState'];
}): JSX.Element {
    const context = React.useMemo<Partial<IColumnScrollContext>>(() => {
        return {
            SELECTORS,
            leftEdgeState: props.leftEdgeState,
            rightEdgeState: props.rightEdgeState,
            isNeedByWidth: true,
        };
    }, []) as IColumnScrollContext;
    return (
        <ColumnScrollContext.Provider value={context}>
            {props.children}
        </ColumnScrollContext.Provider>
    );
}

describe('ControlsUnit/columnScrollReact/ShadowsComponent', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        container.remove();
        container = null;
    });

    describe('Snapshots with different shadows visibility.', () => {
        test.each`
            isLeftVisible | isRightVisible
            ${true}       | ${true}
            ${true}       | ${false}
            ${false}      | ${true}
            ${false}      | ${false}
        `(
            'isLeftVisible=$isLeftVisible, isRightVisible=$isRightVisible',
            ({ isLeftVisible, isRightVisible }) => {
                const { asFragment } = renderAnyComponent(
                    <ShadowsComponent
                        isLeftVisible={isLeftVisible}
                        isRightVisible={isRightVisible}
                    />,
                    { container }
                );
                expect(asFragment()).toMatchSnapshot();
            }
        );
    });

    describe('Snapshots with different shadows user classes.', () => {
        test.each`
            leftShadowClassName   | rightShadowClassName
            ${null}               | ${null}
            ${undefined}          | ${undefined}
            ${' '}                | ${' '}
            ${'custom_className'} | ${undefined}
            ${undefined}          | ${'custom_className'}
            ${'custom_className'} | ${'custom_className'}
        `(
            'leftShadowClassName=$leftShadowClassName, rightShadowClassName=$rightShadowClassName',
            ({ leftShadowClassName, rightShadowClassName }) => {
                const { asFragment } = renderAnyComponent(
                    <ShadowsComponent
                        leftShadowClassName={leftShadowClassName}
                        rightShadowClassName={rightShadowClassName}
                        isLeftVisible={false}
                        isRightVisible={false}
                    />,
                    { container }
                );
                expect(asFragment()).toMatchSnapshot();
            }
        );
    });

    describe('with context and consumer.', () => {
        describe('shadows for different edges state(only valid edges state variants).', () => {
            const testCases = [
                [EdgeState.Visible, EdgeState.Invisible],
                [EdgeState.Invisible, EdgeState.Invisible],
                [EdgeState.Invisible, EdgeState.Visible],
            ].concat(
                AnimationSteps.map((step) => {
                    return [step.leftEdgeState, step.rightEdgeState];
                })
            );

            testCases.forEach(([leftEdgeState, rightEdgeState]) => {
                it(`leftEdgeState=${leftEdgeState}, rightEdgeState=${rightEdgeState}`, () => {
                    const { asFragment } = renderAnyComponent(
                        <FakeContextProviderForShadows
                            leftEdgeState={leftEdgeState}
                            rightEdgeState={rightEdgeState}
                        >
                            <ShadowsComponentConsumer />
                        </FakeContextProviderForShadows>,
                        { container }
                    );
                    expect(asFragment()).toMatchSnapshot();
                });
            });
        });
    });
});
