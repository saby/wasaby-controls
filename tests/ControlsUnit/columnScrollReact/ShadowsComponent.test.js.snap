// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsUnit/columnScrollReact/ShadowsComponent Snapshots with different shadows user classes. leftShadowClassName= , rightShadowClassName=  1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent Snapshots with different shadows user classes. leftShadowClassName=custom_className, rightShadowClassName=custom_className 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left custom_className"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right custom_className"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent Snapshots with different shadows user classes. leftShadowClassName=custom_className, rightShadowClassName=undefined 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left custom_className"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent Snapshots with different shadows user classes. leftShadowClassName=null, rightShadowClassName=null 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent Snapshots with different shadows user classes. leftShadowClassName=undefined, rightShadowClassName=custom_className 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right custom_className"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent Snapshots with different shadows user classes. leftShadowClassName=undefined, rightShadowClassName=undefined 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent Snapshots with different shadows visibility. isLeftVisible=false, isRightVisible=false 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent Snapshots with different shadows visibility. isLeftVisible=false, isRightVisible=true 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent Snapshots with different shadows visibility. isLeftVisible=true, isRightVisible=false 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent Snapshots with different shadows visibility. isLeftVisible=true, isRightVisible=true 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent with context and consumer. shadows for different edges state(only valid edges state variants). leftEdgeState=AnimatedFromInvisibleToVisible, rightEdgeState=AnimatedInvisible 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left controls-ColumnScrollReact__shadow_visible js-controls-ColumnScrollReact_offsetForFixed_guid123"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent with context and consumer. shadows for different edges state(only valid edges state variants). leftEdgeState=AnimatedFromVisibleToInvisible, rightEdgeState=AnimatedInvisible 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left controls-ColumnScrollReact__shadow_visible js-controls-ColumnScrollReact_offsetForFixed_guid123"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent with context and consumer. shadows for different edges state(only valid edges state variants). leftEdgeState=AnimatedInvisible, rightEdgeState=AnimatedFromInvisibleToVisible 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left controls-ColumnScrollReact__shadow_visible js-controls-ColumnScrollReact_offsetForFixed_guid123"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent with context and consumer. shadows for different edges state(only valid edges state variants). leftEdgeState=AnimatedInvisible, rightEdgeState=AnimatedFromVisibleToInvisible 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left controls-ColumnScrollReact__shadow_visible js-controls-ColumnScrollReact_offsetForFixed_guid123"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent with context and consumer. shadows for different edges state(only valid edges state variants). leftEdgeState=AnimatedInvisible, rightEdgeState=AnimatedInvisible 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left controls-ColumnScrollReact__shadow_visible js-controls-ColumnScrollReact_offsetForFixed_guid123"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent with context and consumer. shadows for different edges state(only valid edges state variants). leftEdgeState=AnimatedInvisible, rightEdgeState=AnimatedInvisible 2`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left controls-ColumnScrollReact__shadow_visible js-controls-ColumnScrollReact_offsetForFixed_guid123"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent with context and consumer. shadows for different edges state(only valid edges state variants). leftEdgeState=Invisible, rightEdgeState=Invisible 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left controls-ColumnScrollReact__shadow_visible js-controls-ColumnScrollReact_offsetForFixed_guid123"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent with context and consumer. shadows for different edges state(only valid edges state variants). leftEdgeState=Invisible, rightEdgeState=Visible 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left controls-ColumnScrollReact__shadow_visible js-controls-ColumnScrollReact_offsetForFixed_guid123"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/ShadowsComponent with context and consumer. shadows for different edges state(only valid edges state variants). leftEdgeState=Visible, rightEdgeState=Invisible 1`] = `
<DocumentFragment>
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_left js-controls-ColumnScrollReact_offsetForFixed_guid123"
    data-qa="ColumnScrollReact__shadow_left"
  />
  <div
    class="controls-ColumnScrollReact__shadow controls-ColumnScrollReact__shadow_right controls-ColumnScrollReact__shadow_visible"
    data-qa="ColumnScrollReact__shadow_right"
  />
</DocumentFragment>
`;
